fairlearn/datasets/synthetic.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Microsoft Corporation and Fairlearn contributors.
+# Licensed under the MIT License.
+import numpy as np
+from sklearn.datasets import make_classification
+
+
+def make_synthetic_dataset(seed=12345):
+    """Create a synthetic dataset.
+
+    Parameters
+    ----------
+    seed : int, default=12345
+        The random number generator seed to use.
+
+    Returns
+    -------
+    (X, y, gender) : tuple of numpy.ndarray
+        X : ndarray
+            The generated samples.
+        y : ndarray
+            Labels for the binary classification.
+        gender : ndarray
+            The sensitive feature label.
+    """
+    if rng is None:
+        rng = np.random.RandomState(seed=42)
+
+    classification_kwargs = {
+        'n_features': 20,
+        'n_informative': 4,
+        'n_classes': 2,
+        'random_state': rng,
+    }
+
+    X_women, y_women = make_classification(
+        n_samples=500,
+        class_sep=1,
+        **classification_kwargs,
+    )
```

Comment:
This can be cleaned up a lot. I'm avoiding early abstractions since I wasn't sure what parameters will be exposed.

Reply:
Fair enough! I do think we should start simple and then extend as needed.